### ISSUE TITLE
Exiting: resource ‘filebeat-7.9.0’ exists, but it is not an alias

### ISSUE THREAD

**Issue by somera** [elastic/beats#20849]
For confirmed bugs, please report:
- Version: 7.9.0
- Operating System: Ubuntu 20.04
- Discuss Forum URL: https://discuss.elastic.co/t/exiting-resource-filebeat-7-9-0-exists-but-it-is-not-an-alias/245745
- Steps to Reproduce: Upgrdate or Install ELK 7.9.0

```
==> filebeat/filebeat <==
2020-08-20T01:42:41.965+0200    INFO    [index-management]      idxmgmt/std.go:274      ILM policy successfully loaded.
2020-08-20T01:42:41.966+0200    INFO    [index-management]      idxmgmt/std.go:407      Set setup.template.name to '{filebeat-7.9.0 {now/d}-000001}' as ILM is enabled.
2020-08-20T01:42:41.966+0200    INFO    [index-management]      idxmgmt/std.go:412      Set setup.template.pattern to 'filebeat-7.9.0-*' as ILM is enabled.
2020-08-20T01:42:41.966+0200    INFO    [index-management]      idxmgmt/std.go:446      Set settings.index.lifecycle.rollover_alias in template to {filebeat-7.9.0 {now/d}-000001} as ILM is enabled.
2020-08-20T01:42:41.966+0200    INFO    [index-management]      idxmgmt/std.go:450      Set settings.index.lifecycle.name in template to {filebeat {"policy":{"phases":{"hot":{"actions":{"rollover":{"max_age":"30d","max_size":"50gb"}}}}}}} as ILM is enabled.
2020-08-20T01:42:41.967+0200    INFO    template/load.go:169    Existing template will be overwritten, as overwrite is enabled.
2020-08-20T01:42:42.689+0200    INFO    template/load.go:109    Try loading template filebeat-7.9.0 to Elasticsearch
2020-08-20T01:42:42.828+0200    INFO    template/load.go:101    template with name 'filebeat-7.9.0' loaded.
2020-08-20T01:42:42.828+0200    INFO    [index-management]      idxmgmt/std.go:298      Loaded index template.
2020-08-20T01:42:42.831+0200    ERROR   instance/beat.go:951    Exiting: resource 'filebeat-7.9.0' exists, but it is not an alias
```

**Comment by mtojek** [comment]
Hi @somera , is there any chance that you modified alias/template setup? It would be great if you share your configuration file.

As a workaround you can try this method: https://discuss.elastic.co/t/exiting-resource-metricbeat-7-4-2-exists-but-it-is-not-an-alias/206685/3

**Comment by somera** [comment]
> Hi @somera , is there any chance that you modified alias/template setup? It would be great if you share your configuration file.
> 
> As a workaround you can try this method: https://discuss.elastic.co/t/exiting-resource-metricbeat-7-4-2-exists-but-it-is-not-an-alias/206685/3

Hi @mtojek I fixed my problem last week. I wanted only create the issue, cause other users has same problems.

**Comment by mtojek** [comment]
I'm glad you solved it. I understand that the problem occurred when you tried to go directly from 7.8.0 to 7.9.0, am I correct?

**Comment by somera** [comment]
> I'm glad you solved it. I understand that the problem occurred when you tried to go directly from 7.8.0 to 7.9.0, am I correct?

Now: first installation 7.0.8 -> update to 7.8.1 -> update to 7.9.0

**Comment by falkowich** [comment]
Hi, 

I did the same today.
I get the same problems on my filbeat agents.

We have 7.5.2 as first installation, then 7.7.0 then today 7.9.0.

On a few busy filebeat agent I have `setup.ilm.enabled: false` now, and then they don't warn anymore.
Going to install a new cluster tomorrow and point my agents to that cluster instead.

--
Regards Falk

**Comment by falkowich** [comment]
If this is interesting for "documentation for others in the future"

Some more info from our /_aliases

```{
    "filebeat-7.7.0": {
        "aliases": {}
    },
    "filebeat-7.6.1": {
        "aliases": {}
    },
    "winlogbeat-7.7.0-2020.08.26-000022": {
        "aliases": {
            "winlogbeat-7.7.0": {
                "is_write_index": false
            }
        }
    },
 etc, etc
```

With one working winlogbeat and two examples of broken filbeats :+1: 

--
Regards Falk

**Comment by mtojek** [comment]
@falkowich Thank you for reference points!

Did you manage to grab any logs around "removing data for disk space"?

I wonder if ILM wasn't disabled during the execution.

**Comment by mtojek** [comment]
Found a similar error (and solution) for APM: https://www.elastic.co/guide/en/apm/server/master/common-problems.html#server-resource-exists-not-alias

**Comment by amitavmohanty01** [comment]
@mtojek please take a look at https://discuss.elastic.co/t/exiting-resource-metricbeat-7-4-2-exists-but-it-is-not-an-alias/206685 as well.

I am currently seeing the same error again.

**Comment by JSkier21** [comment]
From what I can tell, there is an issue with legacy and new templates. Once I set these to true in the filebeat.yml, the problem resolved right away. I believe the manual template method will work was well, but I have not tried that. 

I did not have a data loss problem, however I did clone first, to avoid any data loss just in case - I recommend the same. 

```
setup.template.settings:
  setup.template.enabled: true
  setup.template.overwrite: true
```
